pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -734,26 +880,30 @@ def add_repository(self, data: RepoConfigurationData):
 
         :param RepoConfigurationData data: a repo configuration
         """
-        # Create a new repository.
-        repo = self._create_repository(data)
+        repositories = libdnf5.repo.RepoQuery(self._base)
+        repositories.filter_id(data.name)
 
         with self._lock:
-            # Remove an existing repository.
-            if repo.id in self._base.repos:
-                self._base.repos.pop(repo.id)
+            if repositories.empty():
+                # Create a new repository.
+                repo = self._create_repository(data)
+            else:
+                # Replace the existing repository with a new one.
+                repo = self._configure_repository(repositories.get(), data)
```

Comment:
Replace may be a bit misleading here and in the doc string given what is actually done internally. Maybe 'update configuration' / 'reconfigure'?

Reply:
Thanks, I changed the comment and the method docstring.